multistream-select/src/protocol/listener.rs

```diff
@@ -178,6 +163,58 @@ where
     }
 }
 
+
+pub enum ListenerFuture<T: AsyncRead + AsyncWrite> {
```

Comment:
I think this should be "hidden" by naming this for example `ListenerFutureInner` and putting it inside a `ListenerFuture`.

Reply:
Could you explain a bit more what you mean by that?